client/components/domains/register-domain-step/index.jsx

```diff
@@ -781,14 +781,20 @@ class RegisterDomainStep extends Component {
 
 	removeUnavailablePremiumDomain = ( domainName ) => {
 		this.setState( ( state ) => {
-			const newPremiumDomains = { ...state.premiumDomains };
-			delete newPremiumDomains[ domainName ];
-			const searchResults = state.searchResults || [];
+			const newPremiumDomains = Object.fromEntries(
+				Object.entries( state.premiumDomains ).filter( ( [ key ] ) => key !== domainName )
+			);
+			if ( Array.isArray( state.searchResults ) ) {
+				const newSearchResults = state.searchResults.filter(
+					( suggestion ) => suggestion.domain_name !== domainName
+				);
+				return {
+					premiumDomains: newPremiumDomains,
+					searchResults: newSearchResults,
+				};
+			}
 			return {
 				premiumDomains: newPremiumDomains,
-				searchResults: searchResults.filter(
-					( suggestion ) => suggestion.domain_name !== domainName
-				),
```

Comment:
Should we keep `searchResults: null` here? For the protocol, you could just do: `searchResults: state.searchResults`.

Reply:
> Should we keep searchResults: null here?

Based [on this comment](https://github.com/Automattic/wp-calypso/pull/99852/files/8492d810b652816fc46fd164e91fffc8c965c97e#r1957770456) I assumed that `null` has its purpose, also I don't know enough about the code, so keeping the number of changes I make sounds safer. 😅

Reply:
Yes, but you're essentially doing `setState` without `searchResults` if it's `null`, which removes `searchResults`. I think we want to keep it instead. Will suggest an improvement for you to consider.